extensions/csv/src/main/java/io/deephaven/csv/densestorage/QueueWriter.java

```diff
@@ -0,0 +1,203 @@
+package io.deephaven.csv.densestorage;
+
+import io.deephaven.csv.containers.CharSlice;
+
+import java.util.function.BiFunction;
+import java.util.function.IntFunction;
+
+public class QueueWriter<TARRAY, TREADER> {
+    private final Object sync;
+    private QueueNode<TARRAY> tail;
+    protected final int blobSize;
+    private final IntFunction<TARRAY> arrayFactory;
+    private final BiFunction<Object, QueueNode<TARRAY>, TREADER> readerFactory;
+    private boolean allowReaderCreation;
+    private TARRAY genericBlock;
+    protected int begin;
+    protected int current;
+    protected int end;
+
+    protected QueueWriter(int blobSize,
+                          IntFunction<TARRAY> arrayFactory,
+                          BiFunction<Object, QueueNode<TARRAY>, TREADER> readerFactory) {
+        this.sync = new Object();
+        // Placeholder object at head of linked list
+        this.tail = new QueueNode<>(null, 0, 0, false);
```

Comment:
Would this be better named as "sentinel" ?

Reply:
I think the "sentinel" term implies some kind of special value that the readers know about and are able to detect. This node isn't really a "sentinel" (or even a "placeholder" for that matter). It is a degenerate, empty block whose existence makes it easier to write code (because the list is never empty and so I don't have to do tests for head == null. The list always has at least this degenerate block in it).

Reply:
I think you just described what a sentinel is supposed to be: https://en.wikipedia.org/wiki/Sentinel_node
In particular, sentinel does no imply anything that is part of the user visible API of the class, is an implementation detail to avoid having to special-case null.  Like, literally.

Reply:
Well, it is also a very weird video game from the 80s.  https://en.wikipedia.org/wiki/The_Sentinel_(video_game)